package.json

```diff
@@ -204,14 +204,14 @@
     "JSONStream": "^1.0.3",
     "balena-config-json": "^4.1.0",
     "balena-device-init": "^5.0.2",
-    "balena-errors": "^4.4.1",
+    "balena-errors": "^4.7.1",
     "balena-image-manager": "^7.0.3",
     "balena-preload": "^10.3.1",
     "balena-release": "^3.0.0",
-    "balena-sdk": "^15.6.0",
+    "balena-sdk": "^15.18.0",
```

Comment:
This bump is welcome because we want and need to keep the CLI updated with changes to the SDK, but it also means that we will have to make some (manual\*) testing of the CLI with reference to the SDK's Changelog:
https://github.com/balena-io/balena-sdk/blob/master/CHANGELOG.md

\* Manual testing because the automated test coverage in the CLI is still limited.

Reply:
Is there a list of manual tests that usually needs to happen? I could run it it that helps.
Or is it about going through the changes of `balena-sdk` and make sure all possible affected areas of the `cli` are working as expected?

Are you planning to add automated tests in relation to these dependencies? It's probably not related to this issue.

I can write an automated test for the current scenario, that would pave the way for other integration tests. This case is tricky, because of the root user situation, but we can still try to mock `fs` itself. Or we could try changing CI to not run as `root`. What do you reckon?

Reply:
It is about going through the changes of `balena-sdk` and making sure all likely affected areas of the CLI are working as expected - sadly. 😞 

Automated tests for the current scenario are welcome, however I have a feeling (unconfirmed) that the current scenario is less likely to fail than other areas of the CLI that may be affected by the `balena-sdk` bump. Which would mean that the best use of time / effort, in relation to testing, would be to write test cases for CLI commands that use the SDK and don't currently have any tests at all. Of course that while this would be welcome, I would not expect you to do this as it is not related to issue #1667 which is the topic of this PR.

Another issue is that [versionbot failed to produce balena-sdk version 15.18.0](https://github.com/balena-io/balena-sdk/pull/1018#issuecomment-729236877). Until it gets sorted, this PR could be tested with the following line in `package.json`:
```
  "balena-sdk": "^15.18.0-josecoelho-update-balena-auth-36282cee7b3f394b690d0d430dc5705c075b67d1",
```

> This case is tricky, because of the root user situation, but we can still try to mock `fs` itself. Or we could try changing CI to not run as `root`. What do you reckon?

I think that mocking `fs` is a better option for now. Changing the CI to not run as `root` would be a large endeavour given our CI infrastructure (macOS and Windows workers run on physical hosts in physical offices, and would require collaboration of other team members, who would be asking why this is higher priority then their current long list of priorities. I would myself struggle to justify it, as the CLI has tens of much higher priority, long standing issues to be resolved...).

Reply:
Can we have another issue to write the missing tests? I might be able to slowly write them up when I get bored :). I believe it's less than ideal to have such a crucial dependency requiring non specified manual tests before an upgrade. It's a big risk to take on if you need to solve a problem that requires and upgrade to the `sdk` quickly.

Reply:
Sure thing! But we cannot merge this PR without at least manually testing the CLI (the non specified manual tests). I was not necessarily expecting you to test the CLI - if you don't, I will. :-)  We need/want to keep the CLI updated with newer SDK versions anyway. Another complication is that openBalena and balenaCloud are not perfectly in sync re API and SDK, and yet we currently have only one CLI that should work for both. I will also ask team members if they are aware of any SDK changes that might behave differently between openBalena and balenaCloud.

Reply:
> _I will also ask team members if they are aware of any SDK changes that might behave differently between openBalena and balenaCloud_

Got an "all clear" from the team re openBalena. It should be enough to test with balenaCloud.

Reply:
I've done some brief manual testing of selected commands that exercise the SDK, using this line in `package.json`:
```
 "balena-sdk": "^15.18.0-josecoelho-update-balena-auth-36282cee7b3f394b690d0d430dc5705c075b67d1",
```
It seem to be all working OK... I was also going through the SDK changelog with a bit more attention, and I was not alarmed by any of the entries. I would be OK with approving this PR, after the proper SDK version is available. I will be chasing the team about the CI failure to generate a new SDK version in its Changelog and npm registry.